examples/md-flexible/src/domainDecomposition/RegularGridDecomposition.cpp

```diff
@@ -114,164 +145,110 @@ std::array<int, 6> RegularGridDecomposition::getExtentOfSubdomain(const int subd
 }
 
-void RegularGridDecomposition::exchangeHaloParticles(SharedAutoPasContainer &autoPasContainer) {
-  for (int i = 0; i < _dimensionCount; ++i) {
-    std::vector<ParticleType> haloParticles{};
-    for (int j = i; j < _dimensionCount; ++j) {
-      const size_t dimensionIndex = j % _dimensionCount;
-
-      // completely bypass Halo particle exchange in this dimension if boundaries in this direction are not periodic
-      // *and* if both local boundaries are the global boundaries in this dimension
-      if (_boundaryType[dimensionIndex] != options::BoundaryTypeOption::periodic and
-          _localBoxMin[dimensionIndex] == _globalBoxMin[dimensionIndex] and
-          _localBoxMax[dimensionIndex] == _globalBoxMax[dimensionIndex])
-        continue;
-
-      std::vector<ParticleType> particlesForLeftNeighbour{};
-      std::vector<ParticleType> particlesForRightNeighbour{};
-
-      if (j == i) {
-        collectHaloParticlesForLeftNeighbour(autoPasContainer, dimensionIndex, particlesForLeftNeighbour);
-        collectHaloParticlesForRightNeighbour(autoPasContainer, dimensionIndex, particlesForRightNeighbour);
-      }
+void RegularGridDecomposition::exchangeHaloParticles(AutoPasType &autoPasContainer) {
+  std::vector<ParticleType> haloParticles{};
 
-      double leftHaloMin = _localBoxMin[dimensionIndex] - _skinWidth;
-      double leftHaloMax = _localBoxMin[dimensionIndex] + _cutoffWidth + _skinWidth;
-      double rightHaloMin = _localBoxMax[dimensionIndex] - _cutoffWidth - _skinWidth;
-      double rightHaloMax = _localBoxMax[dimensionIndex] + _skinWidth;
-
-      for (const auto &particle : haloParticles) {
-        auto position = particle.getR();
-
-        // check left boundary is a periodic (global) boundary
-        if (_boundaryType[dimensionIndex] == options::BoundaryTypeOption::periodic and
-            _localBoxMin[dimensionIndex] == _globalBoxMin[dimensionIndex]) {
-          if (position[dimensionIndex] >= leftHaloMin and position[dimensionIndex] < leftHaloMax) {
-            particlesForLeftNeighbour.push_back(particle);
-
-            // Apply boundary condition
-            if (_localBoxMin[dimensionIndex] == _globalBoxMin[dimensionIndex]) {
-              position[dimensionIndex] =
-                  position[dimensionIndex] + (_globalBoxMax[dimensionIndex] - _globalBoxMin[dimensionIndex]);
-              particlesForLeftNeighbour.back().setR(position);
-            }
-          }
-        }
+  for (int dimensionIndex = 0; dimensionIndex < _dimensionCount; ++dimensionIndex) {
+    // completely bypass Halo particle exchange in this dimension if boundaries in this direction are not periodic
+    // *and* if both local boundaries are the global boundaries in this dimension
+    if (_boundaryType[dimensionIndex] != options::BoundaryTypeOption::periodic and
+        _localBoxMin[dimensionIndex] == _globalBoxMin[dimensionIndex] and
+        _localBoxMax[dimensionIndex] == _globalBoxMax[dimensionIndex])
+      continue;
```

Comment:
```suggestion
        _localBoxMax[dimensionIndex] == _globalBoxMax[dimensionIndex]) {
      continue;
    }
```

No strong opinion - just think it is clearer if `continue;` on different line